File: evidence/generated-systems/angular/governance-on-angular/src/app/models/ComplianceProgram.js

// Define collection and schema for ComplianceProgram
export interface ComplianceProgram {
    name:
	type : string
    framework:
	type : string
    Organization:
	type : Schema.Types.ObjectId
    Requirements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ComplianceRequirement' }]
    Controls:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Control' }]
    Attestations:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Attestation' }]
    Regulations:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Regulation' }]
    Status:
 	type : String
#
    collection: 'compliancePrograms'
}

File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/ComplianceProgram.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// ComplianceProgram Declaration
//==============================================================
type ComplianceProgram struct {
    gorm.Model
     Name                                    string
    Framework                                    string
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
     Requirements           []ComplianceRequirement `gorm:"foreignKey:RequirementsFromComplianceProgramId"`
     Controls           []Control `gorm:"foreignKey:ControlsFromComplianceProgramId"`
     Attestations           []Attestation `gorm:"foreignKey:AttestationsFromComplianceProgramId"`
     Regulations           []Regulation `gorm:"foreignKey:RegulationsFromComplianceProgramId"`
    Status                      ComplianceStatus

// parent associations as their child

}

